### PR TITLE
nixVersions.unstable: 2.14.1 -> 2.15.0

### PR DESCRIPTION
**Pull request by Mindavi**
###### Description of changes

https://github.com/NixOS/nix/blob/master/doc/manual/src/release-notes/rl-2.15.md

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

This could definitely benefit from some more testing, but opening this already to get the discussion started.

### CODE REVIEW

**Comment by Kranzes** [comment]
Meanwhile nix stable on unstable is still 2.13...

**Comment by zowoq** [comment]
Should be two commits:

nixVersions.unstable: init at 2.15.0
nixVersions.unstable: 2.14 -> 2.15

File: pkgs/top-level/config.nix [hunk 1]
```diff
@@ -61,6 +61,7 @@ let
 
     contentAddressedByDefault = mkMassRebuild {
       feature = "set `__contentAddressed` to true by default";
+      #default = true;
```
Comment (zowoq) [inline pkgs/top-level/config.nix:64]: Committed by accident?

Comment (Mindavi) [reply pkgs/top-level/config.nix:64]: How did I manage to do that? Yeah, accident.

**Comment by Ma27** [comment]
Can we also backport the `nix_2_15` attribute to stable, please? The default nix version not, of course %)

**Pull request merged**